### DISCUSSION TITLE
How do I get the deployment environment?

### DISCUSSION THREAD

**Discussion by shi-yan** [community/community#26514] [Actions]
<p>I want to do auto deployment using github Action, but I don’t know how to read the deployment enviroment info within github Action.</p>
<p>We have 2 environments, one is the production environment, the other is a testing environment. I want the user to be able to choose which one to deploy and which release version or commit ref id to deploy. (in case the user wants to revert).</p>
<p>I can trigger an Action with the deployment API. But I don’t know how to pass the deployment info to the action?</p>

**Comment by shinny-hongyan** [comment, marked as answer]
<p>may be u can use this:</p>
<pre><code>on: push

jobs:
  one:
    runs-on: ubuntu-16.04
    steps:
      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
      - name: Dump job context
        env:
          JOB_CONTEXT: ${{ toJson(job) }}
        run: echo "$JOB_CONTEXT"
      - name: Dump steps context
        env:
          STEPS_CONTEXT: ${{ toJson(steps) }}
        run: echo "$STEPS_CONTEXT"
      - name: Dump runner context
        env:
          RUNNER_CONTEXT: ${{ toJson(runner) }}
        run: echo "$RUNNER_CONTEXT"
      - name: Dump strategy context
        env:
          STRATEGY_CONTEXT: ${{ toJson(strategy) }}
        run: echo "$STRATEGY_CONTEXT"
      - name: Dump matrix context
        env:
          MATRIX_CONTEXT: ${{ toJson(matrix) }}
        run: echo "$MATRIX_CONTEXT"
</code></pre>
<p> it can give u eveything about the env info with a object</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/shi-yan">@shi-yan</a> ,</p>
<p>After the workflow triggered, most info can be got from GITHUB CONTEXT, please refer to the offical <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions" rel="nofollow noopener">doc</a> for more details.</p>
<p>I’m not sure 100% follow, are you wondering how to transfer data to workflow? For deployment issue, when you create the rest api to tigger the workflow, you can add ‘data’ which contains the payload, eg curl command:</p>
<pre><code>curl -X POST .... --data '{"ref": "topic-branch", "payload": {"deploy": "migrate"},"}'
</code></pre>
<p>And get the payload info in workflow via github context with format ${{ github.event.deployment.ref }}.</p>
<p>For action, typically use ‘with’ if the action has defined, please check official <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepswith" rel="nofollow noopener">doc</a> for more details.</p>
<p>Please let me know if it answers, thanks.</p>

**Comment by shi-yan** [comment]
<p>Thank you, yes, this works!</p>

**Comment by shi-yan** [comment]
<p>Yes, I was asking about github context. I didn’t know it exists. Thank you.</p>
